lib/ansible/modules/assemble.py

```diff
@@ -218,7 +218,7 @@ def main():
     path_hash = None
     dest_hash = None
     src = module.params['src']
-    dest = module.params['dest']
+    dest = os.path.abspath(module.params['dest'])
```

Comment:
Instead of expecting individual modules to convert paths to absolute, we should incorporate the check into the validation for `path` type module parameters. A deprecation warning can be issued there, and later converted to an error. Once we have controller-side argument processing, it will be able to perform the relative path check.

Having a deprecation warning on `atomic_move` may still be a good idea, in case there are other direct callers.

Reply:
agreed the conversion belongs in validation, but I did not want to change it for many other modules that use that as a type but do not require abspath. Also not something I would resolve on controller since paths can differ.

Reply:
I wasn't suggesting the controller would be able to resolve the path, just to error out if it's not absolute.

Reply:
We should avoid calling `abspath` on these paths, since that will prevent the warning from being shown. Users need to know that the path(s) they have provided may not work as intended. The current working directly isn't always predictable, such as when dropping privileges on macOS, so converting to absolute here just hides the issue.